notebooks/docs.clj

```diff
@@ -0,0 +1,347 @@
+(clerk/html [:div "As Clojurians we " [:em "really"] " enjoy hiccup"])
+
+;; Alternatively you can pass it an HTML string.
+(clerk/html "Never <strong>forget</strong>.")
+
+;; ### 🔢 Tables
+;; The table viewer api take a number of formats. Each viewer also takes an optional map as a first argument for customization.
+(clerk/table {::clerk/width :full} (into (sorted-map) (map (fn [c] [(keyword (str c)) (shuffle (range 5))])) "abcdefghiklmno"))
+
+;; ### 📒 Markdown
+;; The Markdown viewer is useful for programmatically generated markdown.
+(clerk/md (clojure.string/join "\n" (map #(str "* Item " (inc %)) (range 3))))
+
+
+;; ### 🧮 TeX
+;; The TeX viewer is built on [KaTeX](https://katex.org/).
+(clerk/tex "f^{\\circ n} = \\underbrace{f \\circ f \\circ \\cdots \\circ f}_{n\\text{ times}}.\\,")
+
+;; ### 🗺 Vega Lite
+(clerk/vl {:width 650 :height 400 :data {:url "https://vega.github.io/vega-datasets/data/us-10m.json"
+                                         :format {:type "topojson" :feature "counties"}}
+           :transform [{:lookup "id" :from {:data {:url "https://vega.github.io/vega-datasets/data/unemployment.tsv"}
+                                            :key "id" :fields ["rate"]}}]
+           :projection {:type "albersUsa"} :mark "geoshape" :encoding {:color {:field "rate" :type "quantitative"}}})
+
+;; ### 📊 Plotly
+(clerk/plotly {:data [{:z [[1 2 3] [3 2 1]] :type "surface"}]})
+
+
+;; ### 🎼 Code
+;; The code viewer uses [clojure-mode](https://nextjournal.github.io/clojure-mode/) for syntax highlighting.
+(clerk/code (macroexpand '(when test
+                            expression-1
+                            expression-2)))
+
+(clerk/code '(ns foo "A great ns" (:require [clojure.string :as str])))
+
+(clerk/code "(defn my-fn\n  \"This is a Doc String\"\n  [args]\n  42)")
+
+;; ### 🤹🏻 Applying Viewers
+
+;; In the examples above, we've used convience helper functions like `clerk/html` or `clerk/plotly` to wrap values in a viewer. If you call this on the REPL, you'll notice a given value gets wrapped in a map under the `:nextjournal/value` key with the viewer being in the `:nextjournal/viewer` key.
+
+;; You can also select a viewer using Clojure metadata in order to avoid Clerk interfering with the value.
+
+^{::clerk/viewer clerk/table}
+(def my-dataset
+  [{:temperature 41.0 :date (java.time.LocalDate/parse "2022-08-01")}
+   {:temperature 39.0 :date (java.time.LocalDate/parse "2022-08-01")}
+   {:temperature 34.0 :date (java.time.LocalDate/parse "2022-08-01")}
+   {:temperature 29.0 :date (java.time.LocalDate/parse "2022-08-01")}])
+
+;; ### 👁 Writing Viewers
+
+;; Our goal with the Clerk viewer api is to _keep the toolbox open_
+;; and let folks change both how things are displayed as well as how things behave. In this notebook, we'll go
+;; through how the viewer api works and how you can change it.
+
+;; Clerk comes with a rich set of default viewers, and this is them.
```

Comment:
I would move this to later after you've introduced the concept of a predicate / default viewer more thoroughly. Else the output of `v/default-viewers` is confusing.